spec/dependabot/update_checkers/elixir/hex_spec.rb

```diff
@@ -187,6 +188,87 @@
       it { is_expected.to be >= Gem::Version.new("1.3.0") }
     end
 
+    context "with a dependency with a private source" do
+      let(:mixfile_body) { fixture("elixir", "mixfiles", "private_package") }
+      let(:lockfile_body) { fixture("elixir", "lockfiles", "private_package") }
+
+      before { `mix hex.organization deauth dependabot` }
+
+      let(:dependency) do
+        Dependabot::Dependency.new(
+          name: "example_package_a",
+          version: "1.0.0",
+          requirements: [
+            {
+              file: "mix.exs",
+              requirement: "~> 1.0.0",
+              groups: [],
+              source: nil
+            }
+          ],
+          package_manager: "hex"
+        )
+      end
+
+      context "with good credentials" do
+        let(:credentials) do
+          [
+            {
+              "host" => "github.com",
+              "username" => "x-access-token",
+              "password" => "token"
+            },
+            {
+              "organization" => "dependabot",
+              "token" => "855f6cbeffc6e14c6a884f0111caff3e"
+            }
+          ]
+        end
+
+        it { is_expected.to eq(Gem::Version.new("1.1.0")) }
+      end
+
+      context "with bad credentials" do
+        let(:credentials) do
+          [
+            {
+              "host" => "github.com",
+              "username" => "x-access-token",
+              "password" => "token"
+            },
+            {
+              "organization" => "dependabot",
+              "token" => "111f6cbeffc6e14c6a884f0111caff3e"
+            }
+          ]
+        end
+
+        it "raises a helpful error" do
+          expect { subject }.
+            to raise_error(Dependabot::PrivateSourceNotReachable)
+        end
+      end
+
+      context "with no credentials" do
+        let(:credentials) do
+          [
+            {
+              "host" => "github.com",
+              "username" => "x-access-token",
+              "password" => "token"
+            }
+          ]
+        end
+
+        # Do not run this! It will hang indefinitely (presumably waiting for
+        # input)
+        # it "raises a helpful error" do
+        #   expect { subject }.
+        #     to raise_error(Dependabot::PrivateSourceNotReachable)
```

Comment:
@stevedomin - any idea how we can prevent mix from asking us for input, and get it to error instead? Perhaps we need to write our own code to check that we have auth details for all organisations, and run that before Mix runs it's check?